opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -319,25 +319,26 @@ mod tests {
 
     impl LogExporter for ReentrantLogExporter {
         async fn export(&self, _batch: LogBatch<'_>) -> OTelSdkResult {
-            // This will cause a deadlock as the export itself creates a log
+            let _suppress = Context::enter_telemetry_suppressed_scope();
+            // Without the suppression above, this will cause a deadlock as the export itself creates a log
             // while still within the lock of the SimpleLogProcessor.
             warn!(name: "my-event-name", target: "reentrant", event_id = 20, user_name = "otel", user_email = "otel@opentelemetry.io");
             Ok(())
         }
     }
 
     #[test]
-    #[ignore = "See issue: https://github.com/open-telemetry/opentelemetry-rust/issues/1745"]
+    #[ignore = "While this test runs fine, this uses global subscriber and does not play well with other tests and hence ignored in CI."]
```

Comment:
Quite a picky nit, but why not leave the issue link in a comment as well?

Reply:
There is no issue to track this, as there is no bug to be fixed. The test is written in such a way that it needs "global_subscriber", and such tests don't play well with others, so not running in CI.

(Irrespective of that, this test is better fit for the Otel sdk tests, will be moving it in follow ups)

Reply:
> There is no issue to track this, as there is no bug to be fixed.

It's okay if we are following up on this right after the PR, if not we should create a tracking issue for this. I just don't want us to keep making further Context related changes and forget about this test not being run in CI.

> (Irrespective of that, this test is better fit for the Otel sdk tests, will be moving it in follow ups)

If it's simple enough, I would suggest you do it in this PR itself since you're already making changes to the SDK project in this PR.

Reply:
Moving to the SDK project tests, as that feels the right place to do that test.